tests/integration/components/frost-list-test.js

```diff
@@ -123,59 +123,284 @@ describe(test.label, function () {
+      `)
+        return wait()
+      })
+      it('item 0 not selected ', function () {
+        expect(this.$($hook('my-list-item-container', {index: 0})).hasClass('is-selected')).to.eql(false)
+      })
+      it('item 1 not selected', function () {
+        expect(this.$($hook('my-list-item-container', {index: 1})).hasClass('is-selected')).to.eql(false)
+      })
+    })
+  })
+
+  describe('supports basic and specific click', function () {
     beforeEach(function () {
       const testItems = A([
-        {
-          id: '1',
-          isSelected: true
-        },
-        {
-          id: '2',
-          isSelected: false
-        }
+        Ember.Object.create({id: '0'}),
+        Ember.Object.create({id: '1'})
       ])
 
       this.set('items', testItems)
+      const testSelectedItems = A([])
+      this.set('selectedItems', testSelectedItems)
+      this.set('onSelectionChange', (selectedItems) => {
+        this.get('selectedItems').setObjects(selectedItems)
+      })
 
       this.render(hbs`
         {{frost-list
           item=(component 'frost-list-item')
           hook='my-list'
           items=items
+          selectedItems=selectedItems
+          onSelectionChange=onSelectionChange
         }}
       `)
+      return wait()
     })
 
-    it('selects pre selected item', function () {
-      return wait().then(() => {
-        expect(
-          this.$($hook('my-list-item-0')).hasClass('is-selected')
-        ).to.eql(true)
+    describe('when doing basic click', function () {
+      beforeEach(function () {
+        this.$(hook('my-list-item', {index: 0})).click()
+        return wait()
+      })
+
+      it('item 0 is selected', function () {
+        expect(this.$($hook('my-list-item-container', {index: 0})).hasClass('is-selected')).to.eql(true)
+      })
+
+      it('item 1 is not selected', function () {
+        expect(this.$($hook('my-list-item-container', {index: 1})).hasClass('is-selected')).to.eql(false)
+      })
+
+      describe('when doing basic click on previous selected item', function () {
+        beforeEach(function () {
+          this.$(hook('my-list-item', {index: 0})).click()
+          return wait()
+        })
+
+        it('item 0 is not selected', function () {
+          expect(this.$($hook('my-list-item-container', {index: 0})).hasClass('is-selected')).to.eql(false)
+        })
+
+        it('item 1 is not selected', function () {
+          expect(this.$($hook('my-list-item-container', {index: 1})).hasClass('is-selected')).to.eql(false)
+        })
       })
     })
 
-    it('does NOT selecte pre selected item', function () {
-      return wait().then(() => {
-        expect(
-          this.$($hook('my-list-item-1')).hasClass('is-selected')
-        ).to.eql(false)
+    describe('when doing basic click and all is selected', function () {
+      beforeEach(function () {
+        this.$($(hook('my-list-item-container', {index: 0})).find(hook('my-list-selection-checkbox'))).click()
```

Comment:
I was unable to get that to work in that template but I did get it to work in 
{{frost-list-item-selection
     hook=(concat hookPrefix '-selection')
    hookQualifiers=(hash index=index)
    model=model
    onSelect=(action '_select')
 }}